### ISSUE TITLE
why use deepcopy in models.py

### ISSUE THREAD

**Issue by JH-Xie** [EthanRosenthal/spacecutter#3]
it's strange for me, why you use  deepcopy in there? I have tested the model can work without deepcopy, is there any reason for use it?

line 77: self.predictor = deepcopy(predictor)

**Comment by EthanRosenthal** [comment]
Good question! This should probably be explained in the code...

I think I mainly wanted to ensure that a model is "fresh" with totally unfitted parameters whenever `OrdinalLogisticModel` is instantiated. I was worried about scenarios like where I wanted to try out two different hyperparameters (see the change in learning rate):

```python
predictor = nn.Linear(num_features, 1)

skorch_model = NeuralNet(
    module=OrdinalLogisticModel,
    module__predictor=predictor,
    module__num_classes=num_classes,
    lr=0.1,
    criterion=CumulativeLinkLoss,
    callbacks=[
        ('ascension', AscensionCallback()),
    ],
)
skorch_model.fit(X, y)

new_skorch_model = NeuralNet(
    module=OrdinalLogisticModel,
    module__predictor=predictor,
    module__num_classes=num_classes,
    learning_rate=1.0,  # Larger learning rate
    criterion=CumulativeLinkLoss,
    callbacks=[
        ('ascension', AscensionCallback()),
    ],
)
new_skorch_model.fit(X, y)
```

In a scenario like the one above, the `predictor` would have already been fitted after the first call to `skorch_model.fit(X, y)`, and this fitted `predictor` would then be passed into `new_skorch_model`. I knew that I would forget to reinstantiate `predictor`, so this was a way to guard against that.

Another solution would be to reset the `predictor` model parameters every time `OrdinalLogisticModel` is instantiated, but `deepcopy` was the lazy, quick solution 🙂

**Comment by JH-Xie** [comment]
> solution

perfect explanation, thanks a lot, I really appreciate your work in ordinal regression

**Comment by EthanRosenthal** [comment]
Thanks!
